### PR TITLE
Frontend: Name filter

### PR DESCRIPTION
**Pull request by kobim**
Connecting the Search Bar and allowing (client-side) filtering by name
search-restaurants

### CODE REVIEW

File: client/src/components/Header/SearchBar.jsx [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
 
-const SearchBar = () =>
-  <input type="text" placeholder="Find a restaurant" className="header__banner__search" />;
+export const SearchBar = ({ name, setName }) => (
+  <input
+    type="text"
+    placeholder="Find a restaurant"
+    onChange={e => setName(e.target.value)}
+    value={name}
+    className="header__banner__search" />
```
Comment (shem8) [inline client/src/components/Header/SearchBar.jsx]: can we use CSS modules (or styled components)?

File: client/src/components/Header/SearchBar.jsx [hunk 2]
```diff
@@ -1,6 +1,30 @@
...
+import { setNameFilter } from '../RestaurantsList/redux';
+import { getNameFilter } from '../RestaurantsList/redux/selectors';
 
-const SearchBar = () =>
-  <input type="text" placeholder="Find a restaurant" className="header__banner__search" />;
+export const SearchBar = ({ name, setName }) => (
+  <input
+    type="text"
+    placeholder="Find a restaurant"
+    onChange={e => setName(e.target.value)}
```
Comment (shem8) [inline client/src/components/Header/SearchBar.jsx]: You're going to create a new function every time the input change, that's usually not such a big deal but here, in a text input that changed a lot- you're going to create a lot.

File: client/src/components/Header/SearchBar.test.jsx [hunk 3]
```diff
@@ -2,18 +2,35 @@ import React from 'react';
...
+
   it('renders an input', () => {
-    const wrapper = shallow(<SearchBar />);
+    const wrapper = shallow(<SearchBar {...baseProps} />);
 
     expect(wrapper.find('input').prop('placeholder')).toMatch(/find a rest/i);
+    expect(wrapper.find('input').prop('value')).toEqual(baseProps.name);
+  });
+
+  it('invokes setName when text is change', () => {
```
Comment (shem8) [inline client/src/components/Header/SearchBar.test.jsx]: Nice!

File: client/src/components/RestaurantsList/index.jsx [hunk 4]
```diff
@@ -40,7 +42,11 @@ RestaurantList.propTypes = {
   fetchRestaurants: PropTypes.func.isRequired,
 };
 
-const mapStateToProps = ({ restaurants }) => ({ ...restaurants });
+const mapStateToProps = state => ({
+  loading: get(state, 'restaurants.loading', false),
```
Comment (shem8) [inline client/src/components/RestaurantsList/index.jsx]: use selectors

File: client/src/components/RestaurantsList/redux/selectors.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { createSelector } from 'reselect';
```
Comment (shem8) [inline client/src/components/RestaurantsList/redux/selectors.js]: not sure what I think about selectors in a different file and all the rest in one file

File: client/src/components/Header/SearchBar.jsx (hunk 1 above)
Comment (kobim) [reply client/src/components/Header/SearchBar.jsx]: not in this project

File: client/src/components/RestaurantsList/index.jsx [hunk 6]
```diff
@@ -40,7 +41,11 @@ RestaurantList.propTypes = {
   fetchRestaurants: PropTypes.func.isRequired,
 };
 
-const mapStateToProps = ({ restaurants }) => ({ ...restaurants });
+const mapStateToProps = state => ({
+  loading: isRestaurantsListLoading(state),
+  list: filterRestaurants(state),
+  error: getRestaurantsLoadError(state),
```
Comment (NoyaArie) [inline client/src/components/RestaurantsList/index.jsx]: getRestaurantsLoadingState?
getFilteredRestaurants?
getRestaurantsError?

**Pull request merged**